cluster/src/main/java/org/apache/iotdb/cluster/log/manage/RaftLogManager.java

```diff
@@ -582,14 +585,30 @@ public void commitTo(long newCommitIndex) throws LogExecutionException {
           .clear();
     }
     try {
-      int currentSize = committedEntryManager.getTotalSize();
-      int deltaSize = entries.size();
-      if (currentSize + deltaSize > maxNumOfLogsInMem) {
-        int sizeToReserveForNew = maxNumOfLogsInMem - deltaSize;
+      boolean needToCompactLog = false;
+      int numToReserveForNew = minNumOfLogsInMem;
+      if (committedEntryManager.getTotalSize() + entries.size() > maxNumOfLogsInMem) {
+        needToCompactLog = true;
+        numToReserveForNew = maxNumOfLogsInMem - entries.size();
+      }
+
```

Comment:
if `entries.size()=2,minNumOfLogsInMem=100,maxNumOfLogsInMem=200`, `committedEntryManager.getTotalSize()=199`, then the calculated `numToReserveForNew =200-2=198?` I think it should be `min(198, 100) = 100`